client/src/components/common/boost/BoostConfigurator.tsx

```diff
@@ -23,17 +25,49 @@ export default function BoostConfigurator({
   onDeleteConfig,
   onUploadConfig,
 }: BoostConfiguratorProps) {
+  const fileInput = createRef<HTMLInputElement>();
+  const [configType, setConfigType] = useState<BoostConfigType>('all');
+
+  // Function to click the hidden file input button (this is a work around to avoid the ugly
+  // UI of the default input file button)
+  const handleClick = (type: BoostConfigType) => {
+    if (fileInput.current) {
+      setConfigType(type);
+      fileInput.current.click();
+    }
+  };
+
+  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // This method gets called even if the user uploads a file and the second time they cancel
+    // to upload, hence the need to check that the files attribute is not an array of 0 length
```

Comment:
Nit: Would be good to be consistent with whether comments go before or after the thing being commented on. I'd prefer before like the previous function.